app/views/shared/_form_errors_summary.html.erb

```diff
@@ -1,12 +1,14 @@
-<% if errors.any? %>
+<% if record.errors.any? %>
   <div class="govuk-error-summary" aria-labelledby="error-summary-title" role="alert" tabindex="-1" data-module="error-summary">
     <h2 class="govuk-error-summary__title">There is a problem</h2>
     <div class="govuk-error-summary__body">
       <ul class="govuk-list govuk-error-summary__list">
-        <% errors.keys.each do |field| %>
+        <% record.errors.keys.each do |field| %>
+          <%# require 'pry'; binding.pry %>
+
           <li>
-            <a href='#personal_details_<%= field %>'>
-              <%= errors.messages[field].first %>
+            <a href='#<%= record.model_name.name.underscore %>_<%= field %>'>
```

Comment:
This functionality's covered in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/commit/4bb749559c390dfbca5f98049bc4a9e9c8322513 , hopefully you won't need to change things in this branch.

Reply:
Cool, I like that you put it on the helper, I think you do not need to pass explicitly the model name anyway.